torch/_dynamo/variables/builder.py

```diff
@@ -1114,7 +1114,15 @@ def wrap_numpy_ndarray(self, value):
         assert isinstance(value, np.ndarray)
 
         source = NumpyTensorSource(self.get_source())
-        tensor_value = torch.as_tensor(value)
+
+        from torch._numpy import _util
+
+        try:
+            tensor_value = _util._coerce_to_tensor(value)
```

Comment:
This sets a precedent of reaching into `torch._numpy` from the dynamo wrapper layer. In this particular case it's not a big deal to roll it back; in general it seems reasonable to try reducing the duplication between these two layers (dynamo wrapping into faketensors/numpyvariables and torch._numpy.ndarray wrapping tensors).

Reply:
Our `torch._numpy` is a tool already used by dynamo, so this is completely fine, and if anything, the right thing to do.